cmd/kubectl-directpv/drives_move.go

```diff
@@ -18,133 +18,161 @@ package main
-			return err
+	if !srcDrive.IsUnschedulable() {
+		eprintf("source drive is not cordoned", true)
+		os.Exit(1)
+	}
+
+	var requiredCapacity int64
+	volumes := []string{}
+	for result := range getVolumesByNames(ctx, srcDrive.GetVolumes(), true) {
+		if result.Err != nil {
+			eprintf(result.Err.Error(), true)
+			os.Exit(1)
 		}
-		if err := move(ctx, sourceDrive, targetDrive); err != nil {
-			return err
+
+		if result.Volume.IsPublished() {
+			eprintf(fmt.Sprintf("cannot move published volume %v", result.Volume.Name), true)
+			os.Exit(1)
 		}
-		targetDrive.Status.Status = directpvtypes.DriveStatusMoving
-		_, err = driveClient.Update(
-			ctx, targetDrive, metav1.UpdateOptions{TypeMeta: types.NewDriveTypeMeta()},
+
+		requiredCapacity += result.Volume.Status.TotalCapacity
+		volumes = append(volumes, result.Volume.Name)
+	}
+
+	if len(volumes) == 0 {
+		eprintf(fmt.Sprintf("No volumes found in source drive %v", src), false)
+		return
+	}
+
+	destDrive, err := client.DriveClient().Get(ctx, dest, metav1.GetOptions{})
+	if err != nil {
+		eprintf(fmt.Sprintf("unable to get destination drive; %v", err), true)
+		os.Exit(1)
+	}
+
+	if destDrive.GetNodeID() != srcDrive.GetNodeID() {
+		eprintf(
+			fmt.Sprintf(
+				"source and destination drives must be in same node; source node %v; desination node %v",
+				srcDrive.GetNodeID(),
+				destDrive.GetNodeID(),
+			),
+			true,
 		)
-		return err
-	}); err != nil {
-		return err
+		os.Exit(1)
 	}
-	return nil
-}
 
-func validateMoveRequest(sourceDrive, targetDrive *types.Drive) error {
-	if !(sourceDrive.IsCordoned() || sourceDrive.IsLost()) || !targetDrive.IsCordoned() {
-		klog.Error("please make sure both the source and target drives are cordoned")
-		return errInvalidState
+	if !destDrive.IsUnschedulable() {
+		eprintf("destination drive is not cordoned", true)
+		os.Exit(1)
 	}
-	if sourceDrive.Status.NodeName != targetDrive.Status.NodeName {
-		klog.Error("both the source and target drives should be from same node")
-		return errInvalidState
+
+	if destDrive.Status.Status != directpvtypes.DriveStatusReady {
+		eprintf("destination drive is not ready state", true)
+		os.Exit(1)
 	}
-	if sourceDrive.Status.AccessTier != targetDrive.Status.AccessTier {
-		klog.Error("the source and target drives does not belong to the same access-tier")
-		return errInvalidState
+
+	if srcDrive.GetAccessTier() != destDrive.GetAccessTier() {
+		eprintf(
+			fmt.Sprintf(
+				"source drive access-tier %v and destination drive access-tier %v are different",
+				srcDrive.GetAccessTier(),
+				destDrive.GetAccessTier(),
+			),
+			true,
+		)
+		os.Exit(1)
 	}
-	return nil
-}
 
-func move(ctx context.Context, sourceDrive, targetDrive *types.Drive) error {
-	selector, err := getDriveNameSelectors([]string{sourceDrive.Name})
-	if err != nil {
-		return err
+	if destDrive.Status.FreeCapacity < requiredCapacity {
+		eprintf(
+			fmt.Sprintf(
+				"insufficient free capacity on destination drive; required=%v free=%v",
```

Comment:
```suggestion
				"insufficient free capacity on destination drive %s; required=%v free=%v",
```

Reply:
As drive ID is used as source/destination drive, it is fine not to show here.